staging/src/k8s.io/kube-aggregator/pkg/controllers/openapiv3/aggregator/aggregator.go

```diff
@@ -312,3 +426,133 @@ func (s *specProxier) register(handler common.PathHandlerByGroupVersion) {
 		/* removedRelease */ "",
 		http.HandlerFunc(s.handleGroupVersion)))
 }
+
+func isHashCurrent(u *neturl.URL, currentETag string) bool {
+	if hash := u.Query().Get("hash"); len(hash) > 0 {
+		// check if hash is current only if requested
+		return hash == currentETag
+	}
+	return true
+}
+
+func computeETag(data []byte) string {
```

Comment:
We should use upstream instead this function but I understand that first https://github.com/kubernetes/kubernetes/blob/387d97605ee81cb8beacad990255439ee464b5f3/staging/src/k8s.io/apiserver/pkg/endpoints/discovery/aggregated/etag.go#L76 needs to be exported. Could you please add a comment about the upstream function and mentioning they should be equal?

Reply:
Added the explanation. Although the implementation is so simple that I am not sure if we should make it public upstream.  Regarding the functionality divergence, the other functions are much a bigger problem (eg MergeSpecsV3)